lib/native/auth.spec.ts

```diff
@@ -57,3 +59,41 @@ test('AWS Signer SigV4 Headers', async () => {
 
     expect(http_request.headers._flatten()).toEqual(SIGV4TEST_SIGNED_HEADERS);
 });
+
+test('AWS Signer SigV4 Request with body', async () => {
+
+    const credentials_provider = native.AwsCredentialsProvider.newStatic(
+        SIGV4TEST_ACCESS_KEY_ID,
+        SIGV4TEST_SECRET_ACCESS_KEY,
+    );
+
+    const signing_config: native.AwsSigningConfig = {
+        algorithm: native.AwsSigningAlgorithm.SigV4,
+	    signature_type: native.AwsSignatureType.HttpRequestViaHeaders,
+        provider: credentials_provider,
+        region: SIGV4TEST_REGION,
+        service: SIGV4TEST_SERVICE,
+        date: new Date(DATE_STR),
+        // signed_body_value: native.AwsSignedBodyValue.UnsignedPayload,
+        signed_body_header: native.AwsSignedBodyHeaderType.None,
```

Comment:
As long as the signed_body_value is not set, we will need to read from the body for the `CanonicalRequest`. And it will lead to an endless loop for reading from input stream.